### PR TITLE
Add edge mixer template.

### PR DESCRIPTION
**Pull request by jeffmendoza**
The edge template can be used by adapters that wish to construct a
mesh graph based on observed traffic edges.

### CODE REVIEW

File: mixer/template/edge/template.proto [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//   sourceWorkloadName: source.workload.name | "Unknown"
+//   sourceWorkloadNamespace: source.workload.namespace | "Unknown"
+//   destinationUid: destination.uid | "Unknown"
+//   destinationOwner: destination.owner | "Unknown"
+//   destinationWorkloadName: destination.workload.name | "Unknown"
+//   destinationWorkloadNamespace: destination.workload.namespace | "Unknown"
+//   apiProtocol: api.protocol | "Unknown"
+//   contextProtocol: context.protocol | "Unknown"
+// ```
+message Template {
```
Comment (douglas-reid) [inline mixer/template/edge/template.proto]: Do we want/need some discussion of edges being between workload instances here?

File: mixer/template/edge/template.proto [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Destination of edge
+  string destination_workload_namespace = 20;
+  string destination_workload_name = 21;
+  string destination_owner = 22;
+  string destination_uid = 23;
+
+  // Protocol used
+  string context_protocol = 30;
+  string api_protocol = 31;
```
Comment (douglas-reid) [inline mixer/template/edge/template.proto]: Do we imagine extending this to include any of the following:
- `api.operation`
- `destination.service.name` ( and/or `destination.service.namespace`)
- `connection.mtls`

File: install/kubernetes/helm/istio/charts/mixer/templates/crds.yaml [hunk 3]
```diff
@@ -432,6 +432,24 @@ spec:
   version: v1alpha2
 ---
 
+kind: CustomResourceDefinition
```
Comment (douglas-reid) [inline install/kubernetes/helm/istio/charts/mixer/templates/crds.yaml]: @guptasu just to confirm, this is still where we are putting new templates, correct?

File: mixer/template/edge/template.proto (hunk 1 above)
Comment (jeffmendoza) [reply mixer/template/edge/template.proto]: I'm not sure that the template dictates what the adapter chooses to constructs the graph based on.

File: mixer/template/edge/template.proto (hunk 2 above)
Comment (jeffmendoza) [reply mixer/template/edge/template.proto]: Those sound reasonable to add, and are likely useful to adapters.

Comment (jeffmendoza) [reply mixer/template/edge/template.proto]: I decided to just add `destination.service.uid`, as that can be used to put service nodes in a graph. The others are simply more data about an edge. They seem useful, but there are many attributes that seem useful, I woudln't know where to draw the line. Any useful edge data can be added to the template in the future without breaking backwards compatibility.

**Comment by jeffmendoza** [comment]
Merge conflicts fixed

**Pull request closed without merging**